controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -510,6 +510,65 @@ class ControllerManager : public rclcpp::Node
   };
 
   SwitchParams switch_params_;
+
+  class ControllerThreadWrapper
+  {
+  public:
+    ControllerThreadWrapper(
+      std::shared_ptr<controller_interface::ControllerInterfaceBase> & controller,
+      int cm_update_rate)
+    : terminated_(false), controller_(controller), thread_{}, cm_update_rate_(cm_update_rate)
+    {
+    }
+
+    ControllerThreadWrapper(const ControllerThreadWrapper & t) = delete;
+    ControllerThreadWrapper(ControllerThreadWrapper && t) = default;
+    ~ControllerThreadWrapper()
+    {
+      terminated_.store(true, std::memory_order_seq_cst);
+      if (thread_.joinable())
+      {
+        thread_.join();
+      }
+    }
+
+    void start() { thread_ = std::thread(&ControllerThreadWrapper::call_controller_update, this); }
+
+    void call_controller_update()
+    {
+      rclcpp::Time previous_time = controller_->get_node()->now();
```

Comment:
I don't see this used anywhere? Why is this here?

```suggestion
```

Reply:
It will be used in the if statement, as per the ros2_control_nodes logic. I can easily be convinced that we shouldn't use the same thing for calculating the next iteration's time here though, it depends on the use case I think.